src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -88,143 +88,184 @@ public InventorySync(@Nonnull final InventorySyncOptions syncOptions) {
-                                        .thenCompose(resolvedDraft -> {
-                                            final InventoryEntryIdentifier draftIdentifier =
-                                                InventoryEntryIdentifier.of(resolvedDraft);
-                                            if (existingInventories.containsKey(draftIdentifier)) {
-                                                final InventoryEntry existingEntry =
-                                                    existingInventories.get(draftIdentifier);
-                                                return buildUpdateActionsAndUpdate(existingEntry, resolvedDraft);
-                                            } else {
-                                                return attemptCreate(inventoryEntryDraft);
-                                            }
-                                        })
-                                        .exceptionally(exception -> {
-                                            final String errorMessage = format(FAILED_TO_RESOLVE_SUPPLY_CHANNEL,
-                                                inventoryEntryDraft.getSku(), exception.getMessage());
-                                            handleError(errorMessage, exception, 1);
-                                            return null;
-                                        }))
-                                .exceptionally(exception -> {
-                                    final String errorMessage = format(FAILED_TO_RESOLVE_CUSTOM_TYPE,
-                                        inventoryEntryDraft.getSku(), exception.getMessage());
-                                    handleError(errorMessage, exception, 1);
-                                    return null;
-                                })
-                                .toCompletableFuture();
+            .thenCompose(draftWithResolvedCustomTypeReference -> referenceResolver
+                .resolveSupplyChannelReference(draftWithResolvedCustomTypeReference)
+                .thenApply(Optional::of)
+                .exceptionally(exception -> {
+                    final String errorMessage = format(FAILED_TO_RESOLVE_SUPPLY_CHANNEL,
+                        inventoryEntryDraft.getSku(), exception.getMessage());
+                    handleError(errorMessage, exception, 1);
+                    return Optional.empty();
+                }))
+            .exceptionally(exception -> {
+                final String errorMessage = format(FAILED_TO_RESOLVE_CUSTOM_TYPE,
+                    inventoryEntryDraft.getSku(), exception.getMessage());
+                handleError(errorMessage, exception, 1);
+                return Optional.empty();
+            })
+            .toCompletableFuture();
     }
 
     /**
-     * Returns a distinct set of SKUs from the supplied list of inventory entry drafts.
+     * Tries to create the new inventory entry or update an old one, depending whether an entry
```

Comment:
`Checks if the {@code resolvedDraft} matches with an old existing inventory entry. If it does, it tries to update it. If it doesn't, it creates it.`